src/test/ui/generator/static-not-unpin.rs

```diff
@@ -0,0 +1,25 @@
+// Copyright 2017 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+// normalize-stderr-test "std::pin::Unpin" -> "std::marker::Unpin"
```

Comment:
I have not been able to figure out why travis shows a different path for this compared to the two machines with rustc I've been able to test on. I attempted to trace through where the diagnostics get the path from, and as far as I can tell everything should be deterministic. The path chosen appears to come from the [`visible_parent_map` built here](https://github.com/rust-lang/rust/blob/653da4fd006c97625247acd7e076d0782cdc149b/src/librustc_metadata/cstore_impl.rs#L314), this is done as a BFS, and should be deterministic within a single crate. Following the order of an items children back from here through the different representations all the way to `libsyntax` there doesn't appear to be anything that should change the order, then in `libsyntax` an items children come out in parse order. `std` re-exports the `core::marker` module before the `core::pin` module, so the path chosen to be displayed should be `std::marker::Pin`.

Parse/declaration order is also consistent with the failures in https://travis-ci.org/rust-lang/rust/jobs/451369080, that test commit forced the child order while constructing `visible_parent_map` to be sorted by name. All the failures were swapping paths that were declared earlier with paths declared later but alphabetically sorted earlier.

This will also become moot soon as https://github.com/rust-lang/rust/issues/55766 is proposing to remove the `std::pin::Unpin` re-export (although, it does plan to add it to the prelude which will then become the preferred path).